File: c++cDoodles/ArithGeo.cpp
#include <iostream>
using namespace std;

int ArithGeo(int arr[]) {


  int arithmetic = arr[1]- arr[0];
  int geometric = arr[1]/arr[0];




  // code goes here
  return arr[0];

}

int main() {

  // keep this function call here
  /* Note: In C++ you first have to initialize an array and set
     it equal to the stdin to test your code with arrays.
     To see how to enter arrays as arguments in C++ scroll down */

  int A[] = gets(stdin);
  cout << ArithGeo(A);
  return 0;

}

File: JavaDoodles/ArithGeo.java
import java.util.*; 
import java.io.*;

class ArithGeo {  
  String ArithGeo(int[] arr) { 
  
    int aritmethic = arr[1] - arr[0];    
    int geometric = arr[1] / arr[0];
    
    boolean arith = true;    
    boolean geo = true;
    
    for (int i = 2; i<arr.length; i++) {
      
      if (aritmethic != (arr[i]-arr[i-1])) 
          arith = false;
      if (geometric != (arr[i]/arr[i-1]))
          geo = false;
           
    }
    
   
  if(arith === true)
    return "arithmetic";    
  else if(geo === true)
    return "geometric";
     
   return "-1";
    
  } 
  
  public static void main (String[] args) {  
    // keep this function call here     
    Scanner  s = new Scanner(System.in);
    ArithGeo c = new ArithGeo();
    System.out.print(c.ArithGeo(new int[] {2, 6, 18, 54} )); 
  }   
  
}
